rundmcmc/template_main.py

```diff
@@ -0,0 +1,189 @@
+# Imports for I/O processing
+
+import json
+import geopandas as gp
+from networkx.readwrite import json_graph
+import functools
+
+
+# Imports for RunDMCMC components
+# You can look at the list of available functions in each
+# corresponding .py file.
+
+from rundmcmc.accept import always_accept
+
+from rundmcmc.chain import MarkovChain
+
+from rundmcmc.make_graph import (add_data_to_graph, construct_graph)
+
+from rundmcmc.partition import Partition
+
+from rundmcmc.proposals import propose_random_flip_no_loops
+
+from rundmcmc.updaters import (Tally, boundary_nodes, cut_edges,
+                               cut_edges_by_part, exterior_boundaries,
+                               perimeters, polsby_popper,
+                               votes_updaters,
+                               interior_boundaries)
+
+from rundmcmc.validity import (L1_reciprocal_polsby_popper,
+                               UpperBound,
+                               Validator, no_vanishing_districts,
+                               refuse_new_splits, single_flip_contiguous,
+                               within_percent_of_ideal_population)
+
+from rundmcmc.scores import (efficiency_gap, mean_median,
+                             mean_thirdian)
+
+from rundmcmc.run import pipe_to_table
+
+from rundmcmc.output import p_value_report
+
+from vis_output import (hist_of_table_scores, trace_of_table_scores)
+
+
+# Input the path to the graph (either JSON or shapefile) and the label column
+graph_path = "./testData/PA_graph_with_data.json"
+unique_label = "wes_id"
+
+
+# Names of graph columns go here
+pop_col = "POP100"
+area_col = "ALAND10"
+district_col = "CD"
+
+
+# This builds a graph
+graph = construct_graph(graph_path)
+
+# Write graph to file
+with open('graph_with_data.json', 'w') as outfile1:
+    outfile1.write(json.dumps(json_graph.node_link_data(graph)))
```

Comment:
This should be `json_graph.adjacency_data(graph)`, I think. I can't remember why, but I had issues when using `node_link_data()`. Maybe someone can chime in and explain.

Reply:
Changed!